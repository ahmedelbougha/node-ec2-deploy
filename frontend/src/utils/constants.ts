import placeHolderImage from '../assets/images/placeholder.png';

export const theme = {
  colors: {
    header: {
      backgroundColor: '#00264d',
      color: '#fff',
    },
    body: {
      backgroundColor: '#e3fffe',
      color: '#00264d',
    },
    footer: {
      backgroundColor: '#00264d',
      color: '#fff',
    },
    card: {
      backgroundColor: '#fbfbfb',
      color: '#000',
    },
  },
  placeHolderImage,
};

export const EventTypes: {
  [key: string]: {
    color: string;
    background: string;
    iconStyle: string;
    extraData: string[];
  };
} = {
  alert_qualified: {
    color: '#fff',
    background: '#961e1e',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  alert_raised: {
    color: '#fff',
    background: '#fa1e1e',
    iconStyle: '#1e96f3',
    extraData: ['alert_severity'],
  },
  catheter_observation: {
    color: '#fff',
    background: '#de6cec',
    iconStyle: '#1e96f3',
    extraData: ['volume_ml'],
  },
  check_in: {
    color: '#fff',
    background: '#01d5aa',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  check_out: {
    color: '#fff',
    background: '#cacf9e',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  concern_raised: {
    color: '#fff',
    background: '#8d528a',
    iconStyle: '#1e96f3',
    extraData: ['severity'],
  },
  fluid_intake_observation: {
    color: '#fff',
    background: '#87bef7',
    iconStyle: '#1e96f3',
    extraData: ['consumed_volume_ml'],
  },
  food_intake_observation: {
    color: '#fff',
    background: '#201cc4',
    iconStyle: '#1e96f3',
    extraData: ['meal', 'note'],
  },
  general_observation: {
    color: '#fff',
    background: '#0e699f',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  incontinence_pad_observation: {
    color: '#fff',
    background: '#ee1f4f',
    iconStyle: '#1e96f3',
    extraData: ['pad_condition'],
  },
  medication_schedule_created: {
    color: '#fff',
    background: '#29a88e',
    iconStyle: '#1e96f3',
    extraData: ['note', 'type', 'dose_size'],
  },
  medication_schedule_updated: {
    color: '#fff',
    background: '#dca534',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  mental_health_observation: {
    color: '#fff',
    background: '#b5df79',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  mood_observation: {
    color: '#fff',
    background: '#5d6a9f',
    iconStyle: '#1e96f3',
    extraData: ['mood'],
  },
  no_medication_observation_received: {
    color: '#fff',
    background: '#91e2e7',
    iconStyle: '#1e96f3',
    extraData: ['medication_type', 'expected_dose_timestamp'],
  },
  physical_health_observation: {
    color: '#fff',
    background: '#322c6b',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  regular_medication_maybe_taken: {
    color: '#fff',
    background: '#506c57',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  regular_medication_not_taken: {
    color: '#fff',
    background: '#698bd5',
    iconStyle: '#1e96f3',
    extraData: ['note', 'medication_type', 'medication_failure_reason'],
  },
  regular_medication_partially_taken: {
    color: '#fff',
    background: '#ec8dda',
    iconStyle: '#1e96f3',
    extraData: ['note', 'medication_type'],
  },
  regular_medication_taken: {
    color: '#fff',
    background: '#398c53',
    iconStyle: '#1e96f3',
    extraData: ['medication_type'],
  },
  task_completed: {
    color: '#fff',
    background: '#28b5bd',
    iconStyle: '#1e96f3',
    extraData: ['task_schedule_note', 'task_definition_description'],
  },
  task_completion_reverted: {
    color: '#fff',
    background: '#23b798',
    iconStyle: '#1e96f3',
    extraData: ['task_schedule_note', 'task_definition_description'],
  },
  task_schedule_created: {
    color: '#fff',
    background: '#ff856f',
    iconStyle: '#1e96f3',
    extraData: ['note'],
  },
  toilet_visit_recorded: {
    color: '#fff',
    background: '#e2a6ea',
    iconStyle: '#1e96f3',
    extraData: ['visit_type', 'visit_count', 'note', 'observed'],
  },
  visit_cancelled: {
    color: '#fff',
    background: '#a5e7fc',
    iconStyle: '#1e96f3',
    extraData: [],
  },
  visit_completed: {
    color: '#fff',
    background: '#4bdcbc',
    iconStyle: '#1e96f3',
    extraData: [],
  },
};
