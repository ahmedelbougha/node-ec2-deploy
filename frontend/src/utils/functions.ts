import { EventTypes } from './constants';

export function buildEventTableEvents() {
  let eventTypes = [];

  // get the main keys of event types in order to use them to display events counts
  // in the EventTable
  const eventTypeKeys = Object.keys(EventTypes);

  for (let i = 0; i < eventTypeKeys.length; i += 2) {
    // build eventTypes array with 2 elements at a time
    // to build the EventTable as 2 events with their count per row
    eventTypes.push([eventTypeKeys[i], eventTypeKeys[i + 1]]);
  }

  return eventTypes;
}
