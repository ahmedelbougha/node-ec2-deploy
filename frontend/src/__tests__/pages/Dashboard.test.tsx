import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../utils/constants';
import DashboardPage from '../../pages/Dashboard';

const router = createBrowserRouter([
  {
    // dashboard page
    path: '/',
    element: <DashboardPage />,
  },
]);

const mockStore = configureStore([]);

describe('Dashboard Page', () => {
  let testStore: any;

  beforeEach(() => {
    testStore = mockStore({
      recipients: {
        summaryRecipient: {
          care_recipient_id: 'some-recipient-id',
          recipient_summary: {
            alert_qualified: '5',
            catheter_observation: '12',
            medication_schedule_created: '10',
          },
        },
        eventsRecipient: [
          {
            id: 'some-event-id',
            care_recipient_id: 'some-recipient-id',
            event_type: 'task_completed',
            timestamp: '2019-05-12T18:54:09.116Z',
            payload: {
              id: 'some-event-id',
              visit_id: 'some-visit-id',
              timestamp: '2019-05-12T18:54:09.116Z',
              event_type: 'task_completed',
              caregiver_id: 'some-giver-id',
              task_instance_id: 'some-instance-id',
              task_schedule_id: 'scheduled-id',
              care_recipient_id: 'some-recipient-id',
              task_definition_id: 'task-definition-id',
              task_schedule_note: 'some note goes here!',
              task_definition_description: 'some description goes here!',
            },
          },
        ],
      },
    });
  });

  it('Dashboard page in initial state', () => {
    testStore = mockStore({
      recipients: {
        data: [],
        summaryRecipient: {},
        eventsRecipient: [],
      },
    });
    const component = renderer.create(
      <Provider store={testStore}>
        <RouterProvider router={router} />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Dashboard page after loading recipient's data", () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Provider store={testStore}>
          <RouterProvider router={router} />
        </Provider>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
