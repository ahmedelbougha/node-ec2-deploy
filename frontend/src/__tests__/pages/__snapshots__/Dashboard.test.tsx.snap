// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Dashboard page after loading recipient's data 1`] = `
Array [
  <h2>
    Client Info
  </h2>,
  <div
    className="sc-bcXHqe iuWKlD"
  >
    <div
      className="sc-dkrFOg hqJEoX"
    >
      <div
        className="sc-jSUZER kEmbCt"
      >
        <div>
          <img
            alt="Mr./Mrs. some-"
            src="placeholder.png"
          />
        </div>
        <div>
          <h2>
            Mr./Mrs. some-
          </h2>
        </div>
      </div>
      <table
        className="sc-gKPRtg eksiXW"
      >
        <tbody>
          <tr>
            <td>
              alert qualified
            </td>
            <td>
              5
            </td>
            <td>
              alert raised
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              catheter observation
            </td>
            <td>
              12
            </td>
            <td>
              check in
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              check out
            </td>
            <td>
              0
            </td>
            <td>
              concern raised
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              fluid intake observation
            </td>
            <td>
              0
            </td>
            <td>
              food intake observation
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              general observation
            </td>
            <td>
              0
            </td>
            <td>
              incontinence pad observation
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              medication schedule created
            </td>
            <td>
              10
            </td>
            <td>
              medication schedule updated
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              mental health observation
            </td>
            <td>
              0
            </td>
            <td>
              mood observation
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              no medication observation received
            </td>
            <td>
              0
            </td>
            <td>
              physical health observation
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              regular medication maybe taken
            </td>
            <td>
              0
            </td>
            <td>
              regular medication not taken
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              regular medication partially taken
            </td>
            <td>
              0
            </td>
            <td>
              regular medication taken
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              task completed
            </td>
            <td>
              0
            </td>
            <td>
              task completion reverted
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              task schedule created
            </td>
            <td>
              0
            </td>
            <td>
              toilet visit recorded
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              visit cancelled
            </td>
            <td>
              0
            </td>
            <td>
              visit completed
            </td>
            <td>
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="vertical-timeline vertical-timeline--animate vertical-timeline--two-columns"
    >
      <div
        className="vertical-timeline-element--work vertical-timeline-element"
        id=""
        style={null}
      >
        <span
          className="vertical-timeline-element-icon is-hidden"
          onClick={null}
          style={
            Object {
              "background": "#28b5bd",
              "color": "#fff",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
        <div
          className="vertical-timeline-element-content is-hidden"
          onClick={null}
          style={
            Object {
              "background": "#28b5bd",
              "color": "#fff",
            }
          }
        >
          <div
            className="vertical-timeline-element-content-arrow"
            style={
              Object {
                "borderRight": "7px solid #28b5bd",
              }
            }
          />
          <h3
            className="vertical-timeline-element-title"
          >
            TASK COMPLETED
          </h3>
          <h4
            className="vertical-timeline-element-subtitle"
          >
            Sun May 12 2019
          </h4>
          <div>
            <p>
              some note goes here!
            </p>
            <p>
              some description goes here!
            </p>
          </div>
          <span
            className="vertical-timeline-element-date"
          >
            
          </span>
        </div>
      </div>
      <div
        className="vertical-timeline-element vertical-timeline-element--no-children"
        id=""
        style={null}
      >
        <span
          className="vertical-timeline-element-icon is-hidden"
          onClick={null}
          style={
            Object {
              "background": "rgb(16, 204, 82)",
              "color": "#fff",
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <div
          className="vertical-timeline-element-content is-hidden"
          onClick={null}
          style={null}
        >
          <div
            className="vertical-timeline-element-content-arrow"
            style={null}
          />
          
          <span
            className="vertical-timeline-element-date"
          >
            
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Dashboard Page Dashboard page in initial state 1`] = `
<div
  className="sc-gswNZR kkvFUJ"
>
  <h2>
    There's no matching care recipients records!
  </h2>
</div>
`;
